src/utils.js

```diff
@@ -64,6 +64,83 @@ CKEDITOR.define( function() {
 		 */
 		isObject: function( obj ) {
 			return typeof obj === 'object' && !!obj;
-		}
+		},
+
+		/**
+		 * A mixin function to be used to implement the static `extend()` method in classes. It allows for easy creation
+		 * of subclasses.
+		 *
+		 * @param {Object} [proto] Extensions to be added to the subclass prototype.
+		 * @param {Object} [statics] Additional static properties to be added to the subclass constructor.
+		 * @returns {Object} When executed as a static method of a class, it returns the new subclass constructor.
+		 */
+		extendMixin: function( proto, statics ) {
+			var that = this;
+			var child = ( proto && proto.hasOwnProperty( 'constructor' ) ) ?
+					proto.constructor :
+					function() {
+						that.apply( this, arguments );
+					};
+
+			// Copy the statics.
+			utils.extend( child, this, statics );
+
+			// Use the same prototype.
+			child.prototype = Object.create( this.prototype );
+
+			// Add the new prototype stuff.
+			if ( proto ) {
+				utils.extend( child.prototype, proto );
+			}
+
+			return child;
+		},
+
+		/**
+		 * Creates a spy function (ala Sinon.js) that can be used to inspect call to it.
+		 *
+		 * The following are the present features:
+		 *
+		 *  * spy.called: property set to `true` if the function has been called at least once.
+		 *
+		 * @returns {Function} The spy function.
+		 */
+		spy: function() {
+			var spy = function() {
+				spy.called = true;
+			};
+
+			return spy;
+		},
+
+		/**
+		 * Returns a unique id. This id is a number (starting from 1) which will never get repeated on successive calls
+		 * to this method.
+		 *
+		 * @returns {Number} A number representing the id.
+		 */
+		uid: ( function() {
+			var next = 1;
+
+			return function() {
+				return next++;
+			};
+		} )()
 	};
+
+	return utils;
 } );
+
+/**
```

Comment:
This is something I'm proposing for documentation purposes only.

Here we're providing the documentation that should be used for all `extend()` uses in our code... the ones mixing in `extendMixin`.

For example, it is used here:
https://github.com/cksource/ckeditor5-core/blob/t/10/src/mvc/model.js#L46

I'm not totally sure that this is the right syntax for it, because we still don't have JSDuck in place here, but we'll be able to check it in the future. The important now is the general idea.